templates/CRM/Contribute/Form/Contribution/ThankYou.tpl

```diff
@@ -133,29 +133,30 @@
 
         {* Recurring contribution / pledge information *}
         {if $is_recur}
+          {capture assign="ts_frequency_unit"}{ts}{$frequency_unit}{/ts}{/capture}
```

Comment:
Avoid sending variables in `ts`: https://docs.civicrm.org/dev/en/latest/translation/#avoid-variables-inside-strings

Reply:
@mlutfy Do you think it's better - we might want to have an external tpl for this : 

{capture assign="ts_frequency_unit"}
  {if $frequency_unit eq 'day'}{ts}day{/ts}
  {elseif $frequency_unit eq 'week'}{ts}week{/ts}
  {elseif $frequency_unit eq 'month'}{ts}week{/ts}
  {elseif $frequency_unit eq 'year'}{ts}year{/ts}
  {else}{$frequency_unit}
  {/if}
{/capture}

Reply:
I guess it's better because it can be extracted for transifex translation